golang/thrift/outbound.go

```diff
@@ -0,0 +1,323 @@
+package thrift
+
+import "bytes"
+import "errors"
+import "github.com/apache/thrift/lib/go/thrift"
+import tchannel "github.com/uber/tchannel/golang"
+import "golang.org/x/net/context"
+import "time"
+
+// NewTChannelOutboundProtocol creates a TChannelOutboundProtocol
+func NewTChannelOutboundProtocol(remoteHostPort, remoteServiceName, remoteProcessorName string) (*TChannelOutboundProtocol, error) {
+	tchannel, err := tchannel.NewChannel("0.0.0.0:0", nil)
```

Comment:
Is the TChannelOutboundProtocol object shared across multiple calls from the client?  This seems bad; the protocol as currently written is specific to a single request (it has references to request-specific data) so if there are multiple concurrent requests you'll end up with corruption.  

What is the scope of a client?